### PR TITLE
fix(tests): make tests work in FF56 or FF57

### LINKED ISSUE #5547: update marionnette/internjs so it can run with beta-57 firefox
Currently we are on 53 generally. 57b4 definitely will not run.

### PR DESCRIPTION
**Pull request by vladikoff**
- [x] Fix `handler` issues with "new window" tests 

cc @shane-tomlinson 

fixes #5547

### CODE REVIEW

File: tests/functional/connect_another_device.js [hunk 1]
```diff
@@ -150,7 +150,7 @@ define([
         .then(fillOutSignIn(email, PASSWORD))
         .then(testElementExists(selectors.CONFIRM_SIGNIN.HEADER))
         .then(openVerificationLinkInNewTab(email, 0, { query }))
-        .switchToWindow('newwindow')
+        .getAllWindowHandles()        .then(handles => {          return this.remote.switchToWindow(handles[1])        })
```
Comment (vladikoff) [inline tests/functional/connect_another_device.js]: will need to clean this up

**Comment by vladikoff** [comment]
Ref: https://github.com/theintern/leadfoot/issues/124

**Comment by shane-tomlinson** [comment]
This deserves a 🥇

**Comment by shane-tomlinson** [comment]
Info using the latest Selenium and geckodriver.

With Firefox 55, I can get the tests to run, but see a bunch of errors.

With Firefox 56, I cannot invoke `crypto.getRandomValues`.

But wait, wait, there's more! Tests have (so far) run fine with Firefox 57 and 58!

I have a minimal testcase for Fx , I'm going to bug the geckodriver folks in the office tomorrow.

**Comment by shane-tomlinson** [comment]
More news, using geckodriver 0.18.0 and Selenium 3.6.0, Fx 56 and 57 work as expected.

**Comment by shane-tomlinson** [comment]
> More news, using geckodriver 0.18.0 and Selenium 3.6.0, Fx 56 and 57 work as expected.

I was wrong, I forgot between the time I did it and now that I edited node-uuid so that it didn't call crypto.getRandomBytes

**Comment by shane-tomlinson** [comment]
Ref https://github.com/mozilla/geckodriver/issues/1001

**Comment by shane-tomlinson** [comment]
Also filed https://bugzilla.mozilla.org/show_bug.cgi?id=1406348

File: tests/intern.js [hunk 2]
```diff
@@ -78,7 +78,7 @@ function (intern, topic, firefoxProfile) {
       // this tells SeleniumTunnel to download geckodriver
       drivers: [{
         name: 'firefox',
-        version: '0.18.0'
+        version: '0.19.0'
```
Comment (vladikoff) [inline tests/intern.js]: oops sorry I missed that :(

**Comment by vladikoff** [comment]
@shane-tomlinson could you rebase this and then we should be ready to land?

TeamCity will also need to be updated to use 56 or 57

**Comment by vladikoff** [comment]
@shane-tomlinson need to check Circle

File: .travis.yml [hunk 3]
```diff
@@ -7,7 +7,7 @@ services:
   - redis-server
 
 addons:
-  firefox: "53.0"
+  firefox: "57.0b5"
```
Comment (vladikoff) [inline .travis.yml]: Yeap 57 should be good!

**Pull request merged**